File: LLM-Translations/GPT-4/codenet/Python/Go/s957164965.go
package main

import (
	"fmt"
	"sort"
)

func main() {
	var X, N int
	fmt.Scan(&X, &N)

	p := make([]int, N)
	for i := 0; i < N; i++ {
		fmt.Scan(&p[i])
	}

	sort.Ints(p)

	candidates := map[int][]int{}
	for i := -1; i < 102; i++ {

		if contains(p, i) {
			continue
		}

		if _, ok := candidates[abs(X-i)]; !ok {
			candidates[abs(X-i)] = []int{i}
		} else {
			candidates[abs(X-i)] = append(candidates[abs(X-i)], i)
		}
	}

	smallest := minKey(candidates)

	minNum := minIntSlice(candidates[smallest])

	fmt.Println(minNum)
}

func abs(x int) int {
	if x < 0 {
		return -x
	}
	return x
}

func contains(s []int, e int) bool {
	for _, a := range s {
		if a == e {
			return true
		}
	}
	return false
}

func minKey(m map[int][]int) int {
	min := 200
	for k := range m {
		if k < min {
			min = k
		}
	}
	return min
}

func minIntSlice(s []int) int {
	min := s[0]
	for _, v := range s {
		if v < min {
			min = v
		}
	}
	return min
}

// 

File: LLM-Translations/GPT-4/codenet/Python/C/s957164965.c

#include <stdio.h>
#include <stdlib.h>

int compare(const void *a, const void *b) {
    return (*(int*)a - *(int*)b);
}

int main() {
    int X, N;
    scanf("%d %d", &X, &N);
    
    int p[N];
    for (int i = 0; i < N; i++) {
        scanf("%d", &p[i]);
    }
    
    qsort(p, N, sizeof(int), compare);
    
    int candidates[104][2];
    for (int i = 0; i < 104; i++) {
        candidates[i][0] = 1000;
        candidates[i][1] = 0;
    }
    
    for (int i = -1; i < 102; i++) {
        int j;
        for (j = 0; j < N; j++) {
            if (p[j] == i) {
                break;
            }
        }
        
        if (j == N) {
            int diff = abs(X - i);
            if (candidates[diff][1] == 0) {
                candidates[diff][0] = i;
                candidates[diff][1] = 1;
            } else {
                candidates[diff][1]++;
            }
        }
    }
    
    int smallest = 1000;
    for (int i = 0; i < 104; i++) {
        if (candidates[i][1] != 0 && i < smallest) {
            smallest = i;
        }
    }
    
    int minNum = candidates[smallest][0];
    printf("%d\n", minNum);
    
    return 0;
}
